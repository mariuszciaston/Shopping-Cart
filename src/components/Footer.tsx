const Footer = () => {
  return (
    <footer className="mx-auto flex w-full max-w-[1024px] items-center justify-center p-8 text-sm">
      Mariusz Ciasto≈Ñ 2025 &nbsp;
      <a
        href="https://github.com/mariuszciaston/"
        target="_blank"
        rel="noopener"
      >
        <i
          className="fab fa-github fa-2x transition-transform duration-500 ease-in-out hover:rotate-1turn hover:scale-125 hover:transform"
          aria-hidden="true"
        ></i>
      </a>
    </footer>
  );
};

export default Footer;
