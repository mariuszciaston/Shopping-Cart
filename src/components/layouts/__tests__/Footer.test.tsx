import { render, screen } from "@testing-library/react";
import Footer from "../Footer";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom";

describe("Footer component", () => {
  it("renders the footer with correct text", () => {
    render(<Footer />);
    const footerText = screen.getByText(/Mariusz CiastoÅ„ 2025/);
    expect(footerText).toBeInTheDocument();
  });

  it("contains a link to GitHub with the correct href", () => {
    render(<Footer />);
    const githubLink = screen.getByRole("link");
    expect(githubLink).toHaveAttribute(
      "href",
      "https://github.com/mariuszciaston/",
    );
  });

  it("contains GitHub link containing icon", () => {
    render(<Footer />);
    const githubLink = screen.getByRole("link");
    const icon = githubLink.querySelector("i");
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveClass("fab", "fa-github", "fa-2x");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
