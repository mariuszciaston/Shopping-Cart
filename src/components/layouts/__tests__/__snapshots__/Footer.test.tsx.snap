// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="mx-auto flex w-full max-w-[1280px] items-center justify-center p-4 text-sm sm:p-8"
  >
    <span>
      Mariusz Ciastoń 2025  
    </span>
    <a
      href="https://github.com/mariuszciaston/"
      rel="noopener"
      target="_blank"
    >
      <i
        aria-hidden="true"
        class="fab fa-github fa-2x transition-transform duration-500 ease-in-out hover:rotate-1turn hover:scale-125 hover:transform"
      />
    </a>
  </footer>
</DocumentFragment>
`;
